src/main/java/de/teamlapen/vampirism/entity/minion/HunterMinionEntity.java

```diff
@@ -149,53 +158,44 @@ public void stopTargeting() {
     protected void onMinionDataReceived(@Nonnull HunterMinionData data) {
         super.onMinionDataReceived(data);
         this.updateAttackGoal();
-    }
-
-    @Override
-    protected void registerAttributes() {
-        super.registerAttributes();
-        this.getAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(BalanceMobProps.mobProps.VAMPIRE_HUNTER_MAX_HEALTH + BalanceMobProps.mobProps.VAMPIRE_HUNTER_MAX_HEALTH_PL * 3);
-        this.getAttribute(SharedMonsterAttributes.ATTACK_DAMAGE).setBaseValue(BalanceMobProps.mobProps.VAMPIRE_HUNTER_ATTACK_DAMAGE + BalanceMobProps.mobProps.VAMPIRE_HUNTER_ATTACK_DAMAGE_PL * 3);
-        this.getAttribute(SharedMonsterAttributes.MOVEMENT_SPEED).setBaseValue(BalanceMobProps.mobProps.VAMPIRE_HUNTER_SPEED);
-    }
-
-    @Override
-    protected void registerData() {
-        super.registerData();
-        this.getDataManager().register(RAISED_ARM, false);
-
+        this.updateAttributes();
     }
 
     @Override
     protected boolean processInteract(PlayerEntity player, Hand hand) {
         if (!this.world.isRemote() && isLord(player) && minionData != null) {
             ItemStack heldItem = player.getHeldItem(hand);
-            if (heldItem.getItem() == ModItems.hunter_minion_upgrade1) {
-                if (this.minionData.level < 1) {
-                    this.minionData.setLevel(1);
-                    if (!player.abilities.isCreativeMode) heldItem.shrink(1);
-                    player.sendMessage(new TranslationTextComponent("text.vampirism.hunter_minion.equipment_upgrade"));
-                    HelperLib.sync(this);
-                } else {
-                    player.sendMessage(new TranslationTextComponent("text.vampirism.hunter_minion.equipment_wrong"));
-                }
-                return true;
-            } else if (heldItem.getItem() == ModItems.hunter_minion_upgrade2) {
-                if (this.minionData.level < 2) {
-                    this.minionData.setLevel(2);
+            if (heldItem.getItem() instanceof MinionUpgradeItem) {
+                int upgradeLevel = ((MinionUpgradeItem) heldItem.getItem()).getLevel();
+                if (this.minionData.level < upgradeLevel) {
+                    this.minionData.level++;
                     if (!player.abilities.isCreativeMode) heldItem.shrink(1);
-                    player.sendMessage(new TranslationTextComponent("text.vampirism.hunter_minion.equipment_upgrade"));
+                    player.sendStatusMessage(new TranslationTextComponent("text.vampirism.hunter_minion.equipment_upgrade"), false);
+                    player.sendStatusMessage(new TranslationTextComponent("text.vampirism.minion.upgraded"), true);
```

Comment:
There should be only one of the status messages, because the message in the actionbar (true) can not be read